std/range/package.d

```diff
@@ -6777,6 +6790,376 @@ if (isForwardRange!Source && hasLength!Source)
     assert(equal(chunks, [[1], [2], [3], [], []]));
 }
 
+/**
+This range iterates over a fixed-sized sliding window (k-mers)
+of size `windowSize` of a `source` range.
+$(D Source) must be a forward range. `windowSize` must be greater than zero.
+For `windowSize = 2` it is similar to `zip(source, source.save.dropOne)`.
+
+Params:
+    r = Range from which the slides will be selected
+    windowSize = Sliding window size (default 2)
+
+See_Also: $(LREF chunks)
+
+Returns: Forward range of all sliding windows with propagated bidirectionality,
+         conditional random access and slicing.
+*/
+auto slides(Source)(Source source, size_t windowSize = 2)
```

Comment:
I'm against the default windowSize. If you use slides you properly have a specific windowSize in mind. It just makes for more documentation and more reading.